library/objective-c/EnvoyNetworkMonitor.m

```diff
@@ -0,0 +1,51 @@
+#import "library/objective-c/EnvoyEngine.h"
+
+#import "library/common/main_interface.h"
+
+#import <SystemConfiguration/SystemConfiguration.h>
+
+@implementation EnvoyNetworkMonitor
+
++ (void)startReachabilityIfNeeded {
+  static dispatch_once_t onceToken;
```

Comment:
nit: this can actually be named something meaningful like `reachabilityStarted`

Reply:
Done